lib/upipe/ubuf_block_mem.c

```diff
@@ -216,6 +225,38 @@ static int ubuf_block_mem_splice(struct ubuf *ubuf, struct ubuf **new_ubuf_p,
     return UBASE_ERR_NONE;
 }
 
+/** @This reallocated the shared buffer.
+ *
+ * @param ubuf pointer to ubuf
+ * @param size new size of the buffer
+ */
+static inline bool ubuf_block_mem_realloc(struct ubuf *ubuf, size_t size)
+{
+    struct ubuf_block_mem *block_mem = ubuf_block_mem_from_ubuf(ubuf);
+    bool ret = umem_realloc(&block_mem->shared->umem, size);
+    if (likely(ret))
+        block_mem->ubuf_block.buffer = ubuf_mem_shared_buffer(block_mem->shared);
+    return ret;
+}
+
+/** @internal @This extends a block_mem ubuf.
+ *
+ * @param ubuf pointer to ubuf
+ * @param new_size required size of the buffer
+ * @return false in case of error, or if the ubuf is shared, or if the operation
+ * is not possible
+ */
+static bool ubuf_block_mem_extend(struct ubuf *ubuf, int new_size)
+{
+    if (!ubuf_block_mem_single(ubuf))
+        return false;
```

Comment:
Be careful, the API has changed, it is now UBASE_ERR_BUSY. See ubuf_pic_resize, which is the most similar function to what you are trying to achieve.